shm/Map.cpp

```diff
@@ -1,9 +1,21 @@
 #include "Map.hpp"
 
+#include <algorithm>
+
 void Map::setCurrentPosition(Island* const currentPosition) {
     currentPosition_ = currentPosition;
 }
 
 void Map::addIsland(const Island& newIsland) {
     islandsLocations_.push_back(newIsland);
 }
+
+Island* Map::getIsland(const Coordinates& coordinate) {
+    auto island = std::find_if(islandsLocations_.begin(), islandsLocations_.end(),
+                               [&coordinate](const auto& isld) {
+                                   return isld.Island::getPosition() == coordinate;
```

Comment:
Do we need to write Island:: ? `isld.getPosition();` is not enough? `isld` is an element from vector of Islands so it should be able to get to getPosition() method, I guess.

Reply:
I thought the same but we don't have Island.cpp where `Island::` should be, we have to implement it yet and I will refactor this to `isld.GetPosition()`